base-theme/assets/webpack/webpack.dev.ts

```diff
@@ -18,7 +23,7 @@ const devOverrides: Configuration = {
   devtool: "eval-source-map",
 
   devServer: {
-    port:    process.env.PORT || 3001,
+    port:    env.WEBPACK_PORT,
```

Comment:
Why not `env.WEBPACK_PORT || 3001`? I get an error if I try to run `yarn test:e2e` without `WEBPACK_PORT` explicitly in my env. We default it to 3001 when running a site for dev, so why wouldn't we do that for tests?

Reply:
> We default it to 3001 when running a site for dev

If we merge this PR as-is, that is not longer true.

But, to be sure, having a required WEBPACK_PORT env variable also annoys me. But I do not think we currently have a good way to set the default, and the current situation on `main` illustrates this.

Currently on main:
- [webpack.dev.ts](https://github.com/mitodl/ocw-hugo-themes/blob/main/base-theme/assets/webpack/webpack.dev.ts#L21) uses `process.env.PORT || 3001` 
- but [base-theme/layout/partials/webpack_url.html](https://github.com/mitodl/ocw-hugo-themes/blob/main/base-theme/layouts/partials/webpack_url.html#L4) always uses `3001`.

So you can see the inconsistency: if you have `PORT` env var set, webpack will respect it but hugo will not. 💥.

Here are some options that occur to me:
1. Hardcode 3001 everywhere. *I don't really want to do this, but it doesn't actually seem terrible*
2. Use an environment variable with no default. If it's not set, you get an error message. *A little annoying.*
3. Use an environment variable with inlined-defaults. *This is the current situation, and IMO it is worse than (1) or (2), because of the possibility for inconsistency, as demonstrated by current `maint` branch.*
4. Somehow have a "single source of truth" for environment variables that also allows setting defaults... I can think of two ways:
    - **4A**. Build a CLI... call it `ocw`. It would be like `./manage.py` in django. We'd use it like:

        ```
        yarn ocw --help
        yarn ocw test
        yarn ocw test-e2e
        yarn ocw build course        <-- build a course; we'd use this in pipelines
        yarn ocw build course --help <-- get help info for all the flags/args to build
        yarn ocw serve course        <-- serve a course in dev mode
        ``` 
    
    where the `ocw` CLI would do something like load environment variables, set default values for some, provide useful error messages, have docstrings...like `manage.py` does.

    **This is appealing to me for a variety of reasons. Setting env variables is one. Another is it's a way to provide documentation without cluttering the README.**
    
    - **4B**. Another option is to have `.env` and `.env.defaults` and

        ```sh
        # replace this
        yarn dotenv <cmd>
        # with this:
        yarn dotenv -e .env -e .env.defaults <cmd>
        ```

        I don't dislike this idea. But then do we have three files? (env, env.defaults, and env.sample)? Or do we just use env.sample in place of env.defaults? I'm not sure all the values in .env.sample are reasonable as defaults.

Reply:
oops, submitted that too earlier...

In summary: 4A is my longer-term preference, but the approach of this PR is (2) which seems ok. Maybe 4B is good.

What do you think? or any other ideas?

Reply:
I actually think 3 is the best option specifically referring to the port. There aren't that many places where it's used to generate inconsistency, and I have never heard of software that spins up a service that binds to a port requiring you to specify said port without a default.

Reply:
> There aren't that many places where it's used to generate inconsistency,

But even with not many, we already have the inconsistency on `main`. Run `PORT=3456 yarn start:course` and observe that OCW does not load webpack assets.

Reply:
Actually, I'm going to try (3)... if I can set all the js defaults in one place, that's good enough. If the env var is needed in Hugo, too, we'll have to set the default in two places. If we can keep it to two places, I think that's ok.

Reply:
However it's accomplished, Node and Hugo don't require you to set a port when you run them and will run on a default port. We shouldn't require that it be set either, but allow it to be overridden. I'd rather hard code 3000 for Hugo and 3001 for Webpack than require the env variable to be set to spin up the server.